src/handlers.ts

```diff
@@ -573,3 +574,80 @@ export async function inboxHandler(
         },
     );
 }
+
+export async function repostAction(
+    ctx: Context<{ Variables: HonoContextVariables }>,
+) {
+    const id = ctx.req.param('id');
+    const apCtx = fedify.createContext(ctx.req.raw as Request, {
+        db: ctx.get('db'),
+        globaldb: ctx.get('globaldb'),
+        logger: ctx.get('logger'),
+    });
+
+    const objectToRepost = await lookupObject(apCtx, id);
+    if (!objectToRepost) {
+        return new Response(null, {
+            status: 404,
+        });
+    }
+
+    const announceId = apCtx.getObjectUri(Announce, {
+        id: createHash('sha256').update(objectToRepost.id!.href).digest('hex'),
+    });
+
+    if (await ctx.get('globaldb').get([announceId.href])) {
+        return new Response(null, {
+            status: 409,
+        });
+    }
+
+    const actor = await apCtx.getActor(ACTOR_DEFAULT_HANDLE); // TODO This should be the actor making the request
+
+    const announce = new Announce({
+        id: announceId,
+        actor: actor,
+        object: objectToRepost,
+        to: PUBLIC_COLLECTION,
+        cc: apCtx.getFollowersUri(ACTOR_DEFAULT_HANDLE),
+    });
+
+    const announceJson = await announce.toJsonLd();
+
+    await ctx.get('globaldb').set([announce.id!.href], announceJson);
+    await addToList(ctx.get('db'), ['reposted'], announce.id!.href);
```

Comment:
@sagzy this is where we will also want to add to the outbox - https://github.com/TryGhost/ActivityPub/blob/main/src/handlers.ts#L273

The `outbox` contains posts created / reposted by the current user

Reply:
Added in a987a82930267c20beedf7c3030e4223e3ead97d